src/core/cuda_common_cuda.cu

```diff
@@ -262,26 +262,21 @@ void gpu_change_number_of_part_to_comm() {
     if (global_part_vars_host.number_of_particles) {
 
       /**pinned memory mode - use special function to get OS-pinned memory*/
-      cuda_safe_mem(cudaHostAlloc((void **)&particle_data_host,
-                                  global_part_vars_host.number_of_particles *
-                                      sizeof(CUDA_particle_data),
-                                  cudaHostAllocWriteCombined));
-      cuda_safe_mem(cudaHostAlloc(
-          (void **)&particle_forces_host,
-          3 * global_part_vars_host.number_of_particles * sizeof(float),
-          cudaHostAllocWriteCombined));
+      cuda_safe_mem(
+          cudaHostAlloc((void **)&particle_data_host,
+                        global_part_vars_host.number_of_particles *
+                            sizeof(CUDA_particle_data),
+                        cudaHostAllocWriteCombined ));
```

Comment:
This should use `cudaHostAllocWriteCombined|cudaHostAllocMapped`. Otherwise the memory copy isn't actually going to be asynchronous.